plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,432 @@
+    if dest_file[0] != "/":
+        dest_file = "/%s" % dest_file
+
+    file_local = src_file.strip()
+
+    file_dir = os.path.dirname(dest_file).strip()
+    file_name = os.path.basename(dest_file)
+    if '.' not in file_name:
+        file_in_iso_path = dest_file.upper() + '.;1'
+    else:
+        file_in_iso_path = dest_file.upper() + ';1'
+
+    if file_dir and file_dir != "/":
+        ret, ret_msg = iso_add_dirs(opened_iso, file_dir)
+        if ret != 0:
+            MODULE_ISO_CUSTOMIZE.fail_json(msg=ret_msg)
+
+    try:
+        # The file will be replaced if it exists in ISO
+        opened_iso.add_file(file_local, iso_path=file_in_iso_path,
+                            rr_name=file_name.lower())
+    except Exception as err:
+        msg = "Failed to add local file %s to ISO with error: %s" % (file_local, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    return 0, ""
+
+
+def iso_delete_file(opened_iso, dest_file):
+    dest_file = dest_file.strip()
+    if dest_file[0] != "/":
+        dest_file = "/%s" % dest_file
+    file_name = os.path.basename(dest_file)
+
+    try:
+        record = opened_iso.get_record(rr_path=dest_file)
+        if record and not record.is_file():
+            return -1, "the file %s does not exists in ISO or not a file." % dest_file
+    except Exception as err:
+        msg = "Failed to get record of file %s with error: %s" % (dest_file, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    if '.' not in file_name:
+        file_in_iso_path = dest_file.upper() + '.;1'
+    else:
+        file_in_iso_path = dest_file.upper() + ';1'
+
+    try:
+        opened_iso.rm_file(
+            iso_path=file_in_iso_path, rr_name=file_name.lower())
+    except Exception as err:
+        msg = "Failed to delete iso file %s with error: %s" % (dest_file, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    return 0, ""
+
+
+def iso_replace_file(opened_iso, src_file, dest_file):
+    ret, msg = iso_delete_file(opened_iso, dest_file)
+    if ret != 0:
+        return ret, msg
+    return iso_add_file(opened_iso, src_file, dest_file)
+
+
+def iso_rebuild(src_iso, dest_iso, op_data_list):
+    try:
+        iso = pycdlib.PyCdlib()
+        iso.open(src_iso)
+
+        for op_data in op_data_list:
+            op = op_data['op'].strip()
+            data = op_data['data']
+            if op == "delete_files":
+                for item in data:
+                    ret, msg = iso_delete_file(iso, item.strip())
+                    if ret != 0:
+                        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+            elif op == "add_files":
+                for item in data:
+                    ret, msg = iso_add_file(iso, item['src_file'],
+                                            item['dest_file'])
+                    if ret != 0:
+                        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+            elif op == "modify_files":
```

Comment:
I would recommend this feature be removed from this module:
* The file is being extracted and re-added for each entry - if multiple modifications are made on the same file, there are better ways to do that.
* If multiple modifications are done on the same file it is possible to alter the same line twice, which could cause a lot of confusion, and debugging *that* within this ISO module would be more challenging than what is needed.
* The code (on line 182) is assuming the file is in UTF-8, which may not be true. And having to deal with charsets in here is probably going to increase the complexity of the module beyond its purpose
* There is a number of different modules and filters to alter contents in Ansible and in this collection. Instead of creating yet another way to change files, it might be a better practice to leave that feature for the specialized tools, and make this module simpler.

Reply:
Thanks. remove this feature.